framework/src/csg/CSGBase.C

```diff
@@ -17,8 +17,78 @@ CSGBase::CSGBase()
 {
 }
 
+CSGBase::CSGBase(const CSGBase & other_base)
+  : _surface_list(other_base.getSurfaceList()),
+    _cell_list(CSGCellList()),
+    _universe_list(CSGUniverseList())
+{
+  // Iterate through all cell references from the other CSGBase instance and
+  // create new CSGCell pointers based on these references. This is done
+  // recursively to properly handle cells with universe fills
+  for (const auto & [name, cell] : other_base.getCellList().getCellListMap())
+    addCellToList(*cell);
+
+  // Link all cells in other_base root universe to current root universe
+  for (auto & root_cell : other_base.getRootUniverse().getAllCells())
+  {
+    const auto & list_cell = _cell_list.getCell(root_cell.get().getName());
```

Comment:
could be an efficiency issue to do everything by name but only time and profiling will tell (no action item now)